test/development/acceptance-app/hydration-error.test.ts

```diff
@@ -7,12 +7,49 @@ import { getRedboxTotalErrorCount } from 'next-test-utils'
 
 // https://github.com/facebook/react/blob/main/packages/react-dom/src/__tests__/ReactDOMHydrationDiff-test.js used as a reference
 
-describe('Error overlay for hydration errors', () => {
+describe('Error overlay for hydration errors in App router', () => {
   const { next, isTurbopack } = nextTestSetup({
     files: new FileRef(path.join(__dirname, 'fixtures', 'default-template')),
     skipStart: true,
   })
 
+  it('includes a React docs link when hydration error does occur', async () => {
```

Comment:
This is really just testing React. But it's interesting since we sometimes inject our own link so it's interesting that we diverge sometimes. Probably best to always replace the react.dev link with https://nextjs.org/docs/messages/react-hydration-error